lib/common/memory.c

```diff
@@ -106,7 +107,7 @@ void h2o_mem_free_recycle(h2o_mem_recycle_t *allocator, void *p)
 void h2o_mem_init_pool(h2o_mem_pool_t *pool)
 {
     pool->chunks = NULL;
-    pool->chunk_offset = sizeof(pool->chunks->bytes);
+    pool->chunk_offset = sizeof(*pool->chunks);
```

Comment:
Would you mind elaborating why the change is necessary? I see that you have changed all the `sizeof`s related to the calculation of the buffer size.

My understanding is that `pool->chunk_offset` has been (and can continue to be) a offset for `(st|un)_h2o_mem_pool_chunk_t::bytes`, and I do not understand why you need to change.

Reply:
because now un_h2o_mem_pool_chunk_t is a union, and if offsetof(un_h2o_mem_pool_chunk_t, bytes) will be 0 which differs from struct.

Reply:
Thank you for the answer.

I have switched the code back to use `sizeof ...bytes`. In addition to being more natural (as stated in my comment above) it does not require the knowledge that 1) `bytes` is the largest member of a 2) union.